v2/internal/controllers/crd_networking_applicationgateway_test.go

```diff
@@ -0,0 +1,389 @@
+			CSqGSIb3DQEJFTEWBBQ1FOpSKUDiMNgnn3GIeT1TJ7imnzAxMCEwCQYFKw4DAhoF
+			AAQUZcDhMEBZrBZVLOTMV1dEBK7cxTcECIAwrio+tqKLAgIIAA==`),
+		},
+	}
+	armid_params := map[string]string{
+		"param-key1": "applicationGateways",
+		"param-val1": appGatewayName,
+		"param-key2": "sslCertificates",
+		"param-val2": subResname,
+	}
+	subResARMID, err := getFrontendPortsARMID(tc, rg, armid_params)
+	tc.Expect(err).To(BeNil())
+	return subRes, subResARMID
+}
+
+func defineApplicationGatewaySslProfile(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup, appGatewayName string) ([]network.ApplicationGatewaySslProfile, string) {
+	subResname := "app-gw-ssl-profile-1"
+	subRes := []network.ApplicationGatewaySslProfile{
+		{
+			Name: to.Ptr[string](subResname),
+			SslPolicy: &network.ApplicationGatewaySslPolicy{
+				//MinProtocolVersion: to.Ptr[network.ProtocolsEnum](network.ProtocolsEnum(network.ProtocolsEnum_STATUS_TLSv1_2)),
+				/*CipherSuites: []network.CipherSuitesEnum{
+					network.CipherSuitesEnum(network.CipherSuitesEnum_TLS_RSA_WITH_AES_256_CBC_SHA256),
+					network.CipherSuitesEnum(network.CipherSuitesEnum_TLS_RSA_WITH_AES_256_CBC_SHA),
+				},*/
+				PolicyName: to.Ptr[network.PolicyNameEnum](network.PolicyNameEnum(network.PolicyNameEnum_AppGwSslPolicy20220101S)),
+				PolicyType: to.Ptr[network.ApplicationGatewaySslPolicy_PolicyType](network.ApplicationGatewaySslPolicy_PolicyType(network.ApplicationGatewaySslPolicy_PolicyType_Predefined)),
+			},
+		},
+	}
+	armid_params := map[string]string{
+		"param-key1": "applicationGateways",
+		"param-val1": appGatewayName,
+		"param-key2": "sslProfiles",
+		"param-val2": subResname,
+	}
+	subResARMID, err := getFrontendPortsARMID(tc, rg, armid_params)
+	tc.Expect(err).To(BeNil())
+	return subRes, subResARMID
+}
+
+func defineApplicationGatewayBackendAddressPool(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup, appGatewayName string) ([]network.ApplicationGatewayBackendAddressPool, string) {
+	subResname := "app-gw-be-pool-1"
+	subRes := []network.ApplicationGatewayBackendAddressPool{
+		{
+			Name: to.Ptr[string](subResname),
+			BackendAddresses: []network.ApplicationGatewayBackendAddress{
+				{
+					IpAddress: to.Ptr[string]("10.0.1.1"),
+				},
+			},
+		},
+	}
+	armid_params := map[string]string{
+		"param-key1": "applicationGateways",
+		"param-val1": appGatewayName,
+		"param-key2": "backendAddressPools",
+		"param-val2": subResname,
+	}
+	subResARMID, err := getFrontendPortsARMID(tc, rg, armid_params)
+	tc.Expect(err).To(BeNil())
+	return subRes, subResARMID
+}
+
+func defineApplicationGatewayBackendHttpSettings(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup, appGatewayName string) ([]network.ApplicationGatewayBackendHttpSettings, string) {
+	subResname := "app-gw-be-pool-1"
+	subRes := []network.ApplicationGatewayBackendHttpSettings{
+		{
+			Name:                           to.Ptr[string](subResname),
```

Comment:
minor: drop `to.Ptr[...]` (the stuff in square brackets) so it's just `to.Ptr(thing)`.